### PR TITLE
test: check misbehavior more independently in p2p_filter.py

### PR DESCRIPTION
**Pull request by robot-visions**
This expands on #18672 in two ways:

- Check positive cases (`filterload` accepted, `filteradd` accepted) in addition to the negative cases added in #18672 
- Address MarcoFalke 's [suggestion](https://github.com/bitcoin/bitcoin/pull/18672#discussion_r412101752) to successfully load a filter before testing `filteradd`

### CODE REVIEW

**Comment by theStack** [comment]
Concept ACK -- nit: now with so many small tests for checking the size limits, you could put them all in a new method of the test?

**Comment by robot-visions** [comment]
Good idea, updated!  And thanks for the review.

File: test/functional/p2p_filter.py [hunk 1]
```diff
@@ -64,19 +64,40 @@ def set_test_params(self):
...
+            filter_node.send_and_ping(msg_filterload(data=b'\xbb'*(MAX_BLOOM_FILTER_SIZE)))
+        filter_node.send_and_ping(msg_filterclear())
+
+        self.log.info('Check that filter with too many hash functions is rejected')
+        with self.nodes[0].assert_debug_log(['Misbehaving']):
+            filter_node.send_and_ping(msg_filterload(data=b'\xaa', nHashFuncs=MAX_BLOOM_HASH_FUNCS+1))
+
+        self.log.info('Check that filter with max hash functions is accepted')
+        with self.nodes[0].assert_debug_log([], unexpected_msgs=['Misbehaving']):
+            filter_node.send_and_ping(msg_filterload(data=b'\xaa', nHashFuncs=MAX_BLOOM_HASH_FUNCS, nTweak=0, nFlags=1))
```
Comment (theStack) [inline test/functional/p2p_filter.py:83]: nit: explicitely setting `nTweak` and `nFlags` is not needed here

**Review by theStack — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/18726/commits/edf2491c733c9063c22e2cb8e04b1c3af874d79e

File: test/functional/p2p_filter.py (hunk 1 above)
Comment (robot-visions) [reply test/functional/p2p_filter.py]: Fixed, thanks!

**Comment by theStack** [comment]
re-ACK https://github.com/bitcoin/bitcoin/commit/cd543d9193ac1882c1b4a8a84e3ac7356a8b7ce9

**Pull request merged**